### ISSUE TITLE
Indended stuff inside POD in a perl file destroyed

### ISSUE THREAD

**Issue by TLINDEN** [Malabarba/aggressive-indent-mode#94]
Hello,

take a look at this file:

http://cpansearch.perl.org/src/TLINDEN/Config-General-2.61/General.pm

I had inserted the line

`A block start and end cannot be on the same line.`

and everything afterwards has been un-indended by aggressive-indent-mode. The problem is, that all that stuff is POD and not code and therefore must be kept as is.

The malformed file is published as you can see from the link, because I didn't notice the change. The line I added were on the bottom of the emacs window and I didn't see the stuff after it.

So, please make sure that documentation sections are being kept as is. And it would also be a great enhancement if the mode could display an indication that it indended stuff (in the minibuffer for example).

Another idea would be to provide your own undo-ring or something, so that I can make a change, see the unwanted auto-indent, undo the auto-indent, but keep my change.

Thanks,
Tom

**Comment by Malabarba** [comment]
Sorry about that. Aggressive-indent avoids indenting inside anything that Emacs identifies as strings or comments. I suppose your issue happened because that documentation is not identified as either.

In order to prevent indenting that region, Emacs needs to know that it's not code. Do you know of any function that could tell us that?

**Comment by Malabarba** [comment]
If there's no way, then we could just add this mode the excluded-modes list so we don't turn on by default.

**Comment by TLINDEN** [comment]
I'm pretty sure, that cperl-mode sets the correct values here. I added the following function to my .emacs for testing:
```
(defun iscomment ()
  (interactive)
  (progn
    (message (format "is-comment: %s" (symbol-value (elt (syntax-ppss) 4))))
    ))
```
Then I open a perl script. If I execute it inside a comment, it correctly prints `is-comment: t` otherwise it prints `is-comment: nil`.

However, after studying your code a little, I saw, that in the end you just call `indent-region`. So I marked a region in a perl script containing a POD section with code samples and executed `indent-region`. And you know what? It messed up the code samples!

Therefore, I'd say it's not aggressive-mode's fault, but cperl-mode. In fact, the problem has already been reported: https://github.com/jrockway/cperl-mode/issues/22.
